sorting/heap_sort.cpp

```diff
@@ -1,52 +1,96 @@
-#include <algorithm>
+/**
+ * \addtogroup sorting Sorting Algorithm
+ * @{
+ * \file
+ * \brief [Heap Sort Algorithm
+ * (HEAP SORT)](https://en.wikipedia.org/wiki/Heapsort) implementation
+ *
+ * \author [Ayaan Khan](http://github.com/ayaankhan98)
+ *
+ * \details
+ *  heapsort is a comparison-based sorting algorithm.
+ *  Heapsort can be thought of as an improved selection sort:
+ *  like selection sort, heapsort divides its input into a sorted
+ *  and an unsorted region, and it iteratively shrinks the unsorted
+ *  region by extracting the largest element from it and inserting
+ *  it into the sorted region. Unlike selection sort,
+ *  heapsort does not waste time with a linear-time scan of the
+ *  unsorted region; rather, heap sort maintains the unsorted region
+ *  in a heap data structure to more quickly find the largest element
+ *  in each step.
+ *
+ *  Time Complexity - O(nlog(n))
+ *
+ */
 #include <iostream>
 
-void heapify(int *a, int i, int n) {
+/**
+ *
+ * Utility Lambda function to print the array after
+ * sorting.
+ *
+ * @param arr array to be printed
+ * @param sz size of array
+ *
+ */
+void printArray(int *arr, int sz) {
```

Comment:
```suggestion
template <typename T>
void printArray(T *arr, int sz) {
```

Reply:
> Suggested change 
> void printArray(int *arr, int sz) {
> template <typename T>
> void printArray(T *arr, int sz) {

Should this function be in camel case `print_array`?

Reply:
> > Suggested change
> > void printArray(int *arr, int sz) {
> > template 
> > void printArray(T *arr, int sz) {
> 
> Should this function be in snake case `print_array`?

Never mind. In c++ google style specs, this is allowed

Reply:
okay if it is allowed then i am leaving this in camel casing